pkg/storaged/content-views.jsx

```diff
@@ -326,7 +345,63 @@
         });
     }
 
+    function repair() {
+        const vgroup = lvol && client.vgroups[lvol.VolumeGroup];
+        if (!vgroup)
+            return;
+
+        const summary = client.lvols_stripe_summary[lvol.path];
+        const missing = summary.reduce((sum, sub) => sum + (sub["/"] ?? 0), 0);
+
+        function usable(pvol) {
+            // must have some free space and not already used for a
+            // subvolume other than those that need to be repaired.
+            return pvol.FreeSize > 0 && !summary.some(sub => !sub["/"] && sub[pvol.path]);
+        }
+
+        const pvs_as_spaces = pvs_to_spaces(client, client.vgroups_pvols[vgroup.path].filter(usable));
+        const available = pvs_as_spaces.reduce((sum, spc) => sum + spc.size, 0);
+
+        if (available < missing) {
+            dialog_open({
+                Title: cockpit.format(_("Unable to repair logical volume $0"), lvol.Name),
+                Body: <p>{cockpit.format(_("There is not enough space available that could be used for a repair. At least $0 are needed on physical volumes that are not already used for this logical volume."),
+                                         utils.fmt_size(missing))}</p>
+            });
+            return;
+        }
+
+        function enough_space(pvs) {
+            const selected = pvs.reduce((sum, pv) => sum + pv.size, 0);
+            if (selected < missing)
+                return cockpit.format(_("An additional $0 must be selected"), utils.fmt_size(missing - selected));
+        }
+
+        dialog_open({
+            Title: cockpit.format(_("Repair logical volume $0"), lvol.Name),
+            Body: <div><p>{cockpit.format(_("Select the physical volumes that should be used to repair the logical volume. At leat $0 are needed."),
+                                          utils.fmt_size(missing))}</p><br /></div>,
+            Fields: [
+                SelectSpaces("pvs", _("Physical Volumes"),
```

Comment:
These 6 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230913-131211-e9281c3e-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)